com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/ProcessConsole.java

```diff
@@ -85,35 +93,76 @@ public void stop() {
         }
     }
 
-    public void onStdout(Consumer<String> callback) {
-        stdoutSubject.subscribe(callback);
-    }
-
-    public void onStderr(Consumer<String> callback) {
-        stderrSubject.subscribe(callback);
+    /**
+     * Register a callback to consume the stdout/stderr message from the process.
+     */
+    public void onData(Consumer<Message> callback) {
+        this.consumer = outputSubject.observeOn(Schedulers.newThread()).subscribe(callback);
```

Comment:
Do we want the onData to be called just once? The second call will overwrite this.consumer and we will lose track of the original consumer reference.